app/scripts/modules/core/src/pipeline/filter/ExecutionFilterModel.ts

```diff
@@ -59,42 +60,26 @@ export class ExecutionFilterModel {
       }
     });
 
-    $rootScope.$on(
-      '$stateChangeStart',
-      (
-        _event,
-        toState: Ng1StateDeclaration,
-        _toParams: StateParams,
-        fromState: Ng1StateDeclaration,
-        fromParams: StateParams,
-      ) => {
-        if (this.movingFromExecutionsState(toState, fromState)) {
-          this.asFilterModel.saveState(fromState, fromParams, mostRecentParams);
-        }
-      },
-    );
-
-    $rootScope.$on(
-      '$stateChangeSuccess',
-      (_event, toState: Ng1StateDeclaration, toParams: StateParams, fromState: Ng1StateDeclaration) => {
-        if (this.movingToExecutionsState(toState) && (!toParams.pipeline || !this.groupCount)) {
-          this.mostRecentApplication = toParams.application;
-          this.assignViewStateFromCache();
-        }
-        if (this.movingToExecutionsState(toState) && this.isExecutionStateOrChild(fromState.name)) {
-          this.asFilterModel.applyParamsToUrl();
-          return;
-        }
-        if (this.movingToExecutionsState(toState)) {
-          if (this.shouldRouteToSavedState(toParams, fromState)) {
-            this.asFilterModel.restoreState(toParams);
-          }
-          if (this.fromApplicationListState(fromState) && !this.asFilterModel.hasSavedState(toParams)) {
-            this.asFilterModel.clearFilters();
-          }
-        }
-      },
-    );
+    // Wire up transition hooks
+    const saveFilterState = (trans: Transition) =>
+      this.asFilterModel.saveState(trans.from(), trans.params('from'), mostRecentParams);
+
+    const applyCachedViewState = (trans: Transition) => {
+      this.mostRecentApplication = trans.params().application;
+      this.assignViewStateFromCache();
+    };
+
+    const updateUrl = () => this.asFilterModel.applyParamsToUrl();
+
+    const restoreSavedState = (trans: Transition) =>
+      this.asFilterModel.hasSavedState(trans.params()) && this.asFilterModel.restoreState(trans.params());
+
+    const transitionService = ReactInjector.$uiRouter.transitionService;
+    const executionsGlob = 'home.**.application.pipelines.executions.**';
+    transitionService.onSuccess({ exiting: executionsGlob }, saveFilterState);
+    transitionService.onSuccess({ to: executionsGlob }, applyCachedViewState);
+    transitionService.onSuccess({ retained: executionsGlob }, updateUrl, { priority: -1 });
```

Comment:
`{ priority: -1 }` causes this to be run AFTER the router updates the URL.  Previously, it was updating the URL before the prior transition had completed, and that was the source of the infinite loop.

Reply:
Nice! It would be good to put a comment in the code here explaining what `{ priority: -1 }` does (I won't remember what it does or why it's there in a month).